### DISCUSSION TITLE
Add a button to set all linked properties in the inspector

### DISCUSSION THREAD

**Discussion by Kartopod** [godotengine/godot-proposals#10707] [Editor]
There are a lot of instances in the editor inspector where there are linked properties whose values you want to set to the same value. At the moment, since they're separate properties, setting them all to the same value individually is very cumbersome.  

For example: In a StyleBoxFlat, there are the Border Width, Corner Radius, Expand Margin and Content Margin properties. 
![image](https://github.com/user-attachments/assets/4c113334-4484-40e4-9eed-4df9723ec676)

Of course, it still makes sense that they should be separate properties, because sometimes you do indeed want to have different values for them, but an input field (Or a button that pops up an input field) to set all the values of a linked property at once would greatly improve the user experience.

**Comment by aXu-AP** [comment]
Agree, this would be a good QoL improvement. Such thing exists for scale, so it should look similiar:
![kuva](https://github.com/user-attachments/assets/8adae736-da10-4346-88cb-103a795862ca)

Another area which would benefit from this: RichTextLabel's font sizes:
![kuva](https://github.com/user-attachments/assets/e7e09b81-ea3e-4671-84b5-38c69206b615)

**Reply by tetrapod00** [reply]
Yeah, I've been wanting something like this for a while now. I agree this is a good default UI to match current behavior in the editor. 

The VectorN linking is implemented here, and in the rest of this file. But that linking only deals with the subproperties of a single property. Doing something similar for multiple properties would have to reimplement similar code elsewhere.

https://github.com/godotengine/godot/blob/97ef3c837263099faf02d8ebafd6c77c94d2aaba/editor/editor_properties_vector.cpp#L236L242

**Comment by Mickeon** [comment]
I've always had the idea that you should be able to select multiple properties/spin sliders in the Inspector by holding CTRL, but in practice I'm not sure how complex that would be.

**Comment by Kartopod** [comment]
@Mickeon Blender's UX for this is extremely good:
- Dragging vertically allows selection of multiple fields, after the first horizontal dragging is detected, vertical selection is locked.
- You can also drag on only a few fields.
- Dragging fields that have different values selects and sets them to the value you started dragging from.
- 'Selecting' fields allows you to manually input the same value to multiple fields. 
![Blender linked field UX](https://github.com/user-attachments/assets/1d2737df-461b-4769-b52a-28d273b7d38f)

**Reply by aXu-AP** [reply]
There's a few things I don't particularly like in Blender's approach:
* It's easy to accidentally move mouse horizontally before you reach all the fields (could be fixed with bigger margins).
* You can select only neighboring elements.

Overall it does feel good and I do use it in Blender. But I like @Mickeon 's suggestion better (it's also likely easier to implement).

**Reply by Kartopod** [reply]
That's fair. Just to throw around my thoughts aswell, I love how effortless it is to drag over all the fields and affect them at once. Pecking and choosing like how @Mickeon suggested would definitely help make the UX better than what it is now, but I feel like there would still be some friction when you want to set *all* the values together.

This is especially the case with the theme properties (eg: Corner Radius) because 90% of the time you want to quickly set *all four* of them to the same value. Having to click each one (imo) would still be very slow.

**Reply by aXu-AP** [reply]
If we implemented multiselect, shift clicking to choose consecutive properties should also work.

**Reply by Kartopod** [reply]
I suppose a good middle ground for that would be to allow shift clicking, as well as shift click + dragging. This would let you peck and choose individual elements, as well as select neighbouring elements quickly

**Comment by michaelharmonart** [comment]
just a note, maya has a system very similar to blender where you can drag and select multiple channels in the channel box and then you can adjust them by then middle mouse dragging left and right*Anywhere* to adjust them.

Animators love this and It's one of the first tools you're taught for animation in maya. Just to throw it in as another possibility. It avoids the accidental adjustment that's possible in the blender style.

**Comment by AndreyCortez** [comment]
For me at least, I think a very intuitive way of doing this would selecting the fields like you select files on the file syetem explorer. Pressing control you can select one by one, pressing shift you select them from top to bottom. Something like this but a check would be needed to see if the types are the same.

**Comment by kaiiboraka** [comment]
One simple solution to changing ALL values that I've thought of in the past is an additional "All" field, that would be editable even from collapse. Attached is a simplified mockup. Changing this value would automatically change each subvalue to match.

![image](https://github.com/user-attachments/assets/d35b0c13-3c5f-4eff-9ffc-cf83d7de2e4a)

Ctrl+Click still has merit for choosing any arbitrary number of fields less than the max, so I'm still a fan of that solution in addition. But this would be a nice first step to solve the problem of laborious multi-selection when you simply want to change them all.

**Reply by aXu-AP** [reply]
This idea could be nice on some curated cases (like the ones demonstrated). Not everywhere though, that would add clutter and be sometimes nonsensical...

**Comment by shoyguer** [comment]
So I'm trying to develop a contribution, and I made 3 versions, and I'd like you guys to take a look and give me your feedback.

Only initials, similar to X and Y for vectors:
<img width="496" height="864" alt="image" src="https://github.com/user-attachments/assets/6a8d55e5-5b24-4d6d-a6d7-b20854900d45" />

Full names as currently are, though colored:
<img width="433" height="881" alt="image" src="https://github.com/user-attachments/assets/189d8f65-0a0a-41a9-a063-d4524d7ed8f2" />

Default theme.
<img width="449" height="872" alt="image" src="https://github.com/user-attachments/assets/acd8573c-6480-43c6-ad64-76bf217afb53" />

**Reply by Calinou** [reply]
Can we use icons that represent sides or corners instead of text? We have something similar for anchor presets already:

<img width="170" height="290" alt="image" src="https://github.com/user-attachments/assets/885e8cd3-1da0-4d6a-92d0-ffc3a89e5754" />

Feel free to reuse the existing anchor preset icons as a placeholder (we can probably design more suitable icons later on).

Also, this would be a good opportunity to list all four sides/corners a single line for compactness (or perhaps 2 lines if the inspector is narrow).

**Reply by shoyguer** [reply]
This is a very good and cool solution!
Sure, I'll implement that. Thanks for the feedback 😁

**Reply by shoyguer** [reply]
In a single line it doesn't have enough space usually (I'm using a QHD monitor, and I need to make the space pretty big so that the numbers have enough space to be visible and look good enough.

<img width="535" height="636" alt="image" src="https://github.com/user-attachments/assets/22bf9c91-37d7-48a6-a53e-8b3ea4f3dad0" />

So I'm splitting them in 2 lines.

<img width="542" height="646" alt="image" src="https://github.com/user-attachments/assets/7e4e17fc-b52b-4bcc-ab13-d9e1ee20e3be" />

And I made sure it will automatically split all of them into their own lines if they don't have enough space.

<img width="260" height="907" alt="image" src="https://github.com/user-attachments/assets/cad55e03-aa7b-4e48-8dcb-d7f139fc01d8" />

**Reply by Calinou** [reply]
Looks good!

When showing "corner" properties in a 2x2 layout, you should probably swap the bottom-left and bottom-right properties so that they appear in their respective corner. This means their location on the UI matches the corner you are editing.

This won't make as much sense for "side" properties though, but the current layout isn't necessarily more adequate for "side" properties anyway. For "side" properties, we could use a diamond design that spans 3 lines as follows:

```
_ T _
L _ R
_ B _
```

Where `T` = top, `L` = left, `R` = right, `B` = bottom, `_` = empty space.

cc @godotengine/usability in case someone has additional thoughts on this.

**Reply by shoyguer** [reply]
<img width="564" height="905" alt="image" src="https://github.com/user-attachments/assets/6aaf5516-34e4-4af9-9d10-5e8ba7d78395" />

Something like this?
Imo the layout doesn't look good when we have spinboxes.

**Reply by Calinou** [reply]
>  Imo the layout doesn't look good when we have spinboxes.

I think this is down to the SpinBoxes being very wide (relative to the numbers these properties typically house, usually in the 0-99 range). It should look better if you can reduce their width.

**Reply by shoyguer** [reply]
<img width="494" height="747" alt="image" src="https://github.com/user-attachments/assets/cbdf46c9-1bf7-4235-a38d-9d0d5a570ab2" />

Wdyt?

**Reply by Calinou** [reply]
> Wdyt?

Looks great now :slightly_smiling_face:

**Reply by shoyguer** [reply]
Opening a PR, either today or tomorrow.
